guides/v2.1/contributor-guide/backward-compatible-development/index.md

```diff
@@ -39,6 +39,12 @@ Mark the method with the `@deprecated` tag instead of removing it.
 
 Continue returning the same results from the method if possible, so the old functionality is preserved.
 
+If you need to remove the existing functionality and you do not need alternative implementation, you SHOULD change the method code to
+
+```php
+trigger_error('Function is deprecated', E_USER_WARNING);
```

Comment:
Deprecation is not what happens in such situation, so I think deprecation warning will be misleading.
I'd recommend real exception or, even better, removal of entire method in such situation. So that it's visible to everyone that there is a breaking change.

Reply:
`trigger_error` is much better than Exception, you can't catch the user error in the catch block. Removing a  method is a backward incompatible implementation and sometimes it can't be implemented, when you have an inheritance